radicle-httpd/src/api/auth.rs

```diff
@@ -16,75 +11,19 @@ impl Serialize for DateTime {
     }
 }
 
-#[derive(Deserialize, Serialize)]
-pub struct AuthRequest {
-    pub message: String,
-    #[serde(deserialize_with = "deserialize_signature")]
-    pub signature: Signature,
-}
-
-fn deserialize_signature<'de, D>(deserializer: D) -> Result<Signature, D::Error>
-where
-    D: serde::de::Deserializer<'de>,
-{
-    let buf = String::deserialize(deserializer)?;
-    Signature::from_str(&buf).map_err(serde::de::Error::custom)
-}
-
 pub enum AuthState {
     Authorized(Session),
     Unauthorized {
-        nonce: String,
-        expiration_time: DateTime,
+        public_key: PublicKey,
+        expires_at: DateTime,
     },
 }
 
 // We copy the implementation of siwe::Message here to derive Serialization and Debug
 #[derive(Clone, Serialize)]
 #[serde(rename_all = "camelCase")]
 pub struct Session {
-    pub domain: String,
-    pub address: H160,
-    pub statement: Option<String>,
-    pub uri: String,
-    pub version: u64,
-    pub chain_id: u64,
-    pub nonce: String,
+    pub public_key: String,
```

Comment:
Can we make this `radicle::crypto::PublicKey` for better type safety?

Reply:
Fixed in ca25ea9ef95615df774d65fe9de31df899a240d7